aws_topology/stackstate_checks/aws_topology/resources/dynamodb.py

```diff
@@ -56,37 +56,31 @@ def collect_table_description(self, table_name):
 
     def construct_table_description(self, table_name):
         return {
-            "Table": {
-                "TableName": table_name,
-                "TableArn": self.agent.create_arn("AWS::DynamoDB::Table", self.location_info, resource_id=table_name),
-            }
+            "TableName": table_name,
+            "TableArn": self.agent.create_arn("AWS::DynamoDB::Table", self.location_info, resource_id=table_name),
         }
 
     def collect_table(self, table_name):
         # If data collection fails, construct the ARN
         data = self.collect_table_description(table_name) or self.construct_table_description(table_name)
-        # If ARN creation fails, skip tag collection
-        if data.get("Table", {}).get("TableArn"):
-            tags = self.collect_tags(data["Table"]["TableArn"])
-        else:
-            tags = []
+        tags = self.collect_tags(data.get("TableArn")) or []
         return TableData(table=data, tags=tags)
 
-    @set_required_access_v2("dynamodb:ListTables")
     def collect_tables(self):
         for table in [
             self.collect_table(table_name)
             for table_name in client_array_operation(self.client, "list_tables", "TableNames")
         ]:
             yield table
 
+    @set_required_access_v2("dynamodb:ListTables")
+    def process_tables(self):
+        for table_data in self.collect_tables():
+            self.process_table(table_data)
```

Comment:
Will this not throw an exception anymore since we were catching earlier?

Reply:
This fixes the exception. It was not being caught in the right way, and was causing the service to fail out completely instead of just stopping the table execution and emitting a warning. The exception handling is in @set_required_access_v2

Reply:
@Eelviny Then I saw this somewhere that decorator was wrongly placed on the fucntion.